mcp_naa/tools/db_tools.py

```diff
@@ -0,0 +1,209 @@
+# mcp_naa/tools/db_tools.py
+from typing import Optional
+from .. import app # Import the initialized agentad
+from .. import database     # Import the database module
+from ..logging_config import get_logger
+import pyodbc
+
+logger = get_logger(__name__)
+
+@app.mcp_agent.tool("Connect-Database")
+def connect_database(server: str, database_name: str, username: Optional[str] = None,
+                     password: Optional[str] = None, trusted_connection: bool = False) -> str:
+    """Connects to a specified MSSQL database server, overriding environment settings."""
+    from ..config import settings
+    
+    logger.info(f"Tool 'Connect-Database' called for {server}/{database_name}")
+    
+    # Close existing connection first
+    database.close_connection()
+    
+    # Store original settings
+    original_server = settings.DB_SERVER
+    original_db_name = settings.DB_NAME
+    original_user = settings.DB_USER
+    original_password = settings.DB_PASSWORD
+    original_windows_auth = settings.DB_USE_WINDOWS_AUTH
+    
+    try:
+        # Temporarily override settings
+        settings.DB_SERVER = server
+        settings.DB_NAME = database_name
+        settings.DB_USER = username
+        settings.DB_PASSWORD = password
+        settings.DB_USE_WINDOWS_AUTH = trusted_connection
```

Comment:
`username` and `password` are `Optional` so if they're not provided we'll override the setting from config (env) to nothing and the connection will fail.

```suggestion
        settings.DB_USER = username if username is not None else setting.DB_USER
        settings.DB_PASSWORD = password if password is not None else setting.DB_PASSWORD
        settings.DB_USE_WINDOWS_AUTH = trusted_connection
```

Reply:
"Updated."